include/ensmallen_bits/demon_adam/demon_adam_update.hpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * @file demon_sgd_update.hpp
+ * @author Marcus Edel
+ *
+ * Implementation of DemonAdam.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_DEMON_ADAM_DEMON_ADAM_UPDATE_HPP
+#define ENSMALLEN_DEMON_ADAM_DEMON_ADAM_UPDATE_HPP
+
+#include <assert.h>
```

Comment:
I imagine this is already included by other headers?  (Also I'm not sure it's needed.  Correct me if I'm wrong.)

Reply:
No `assert` in https://github.com/mlpack/ensmallen/blob/master/include/ensmallen.hpp.